src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php

```diff
@@ -82,8 +114,13 @@ public function getTypeGuesser()
             $this->guesserLoaded = true;
             $guessers = array();
 
-            foreach ($this->guesserServiceIds as $serviceId) {
-                $guessers[] = $this->container->get($serviceId);
+            foreach ($this->guesserServices as $serviceId => $service) {
+                if (is_string($service)) {
+                    $serviceId = $service;
+                    $service = $this->typeContainer->get($serviceId);
+                }
+
+                $guessers[] = $service;
```

Comment:
If fact, guessers do not need to be instantiated at this point. 
I think we should register `FormTypeGuesserChain` as a service, pass it guessers as an IteratorArgument and remove both `$guesserServices` and `$guesserServiceIds` in favor of a `FormTypeGuesserInterface $guesser` here. Not sure how it would be hard to keep it BC though